### ISSUE TITLE
keras LSTM model - a tf 1.15 equivalent that works with tflite

### ISSUE THREAD

**Issue by yonatanbitton** [tensorflow/tensorflow#46948]
Hello

**TLDR**: How to implement this model using `tf.lite.experimental.nn.TFLiteLSTMCell, tf.lite.experimental.nn.dynamic_rnn`instead `keras.layers.LSTM`?
![image](https://user-images.githubusercontent.com/26148975/107017531-eab71b00-67a7-11eb-9a92-b161b49f439b.png)

I have this network in keras:
```python
inputs = keras.Input(shape=(1, 52))
state_1_h = keras.Input(shape=(200,))
state_1_c = keras.Input(shape=(200,))
x1, state_1_h_out, state_1_c_out = layers.LSTM(200, return_sequences=True, input_shape=(sequence_length, 52),
                                               return_state=True)(inputs, initial_state=[state_1_h, state_1_c])
output = layers.Dense(13)(x1)

model = keras.Model([inputs, state_1_h, state_1_c],
                    [output, state_1_h_out, state_1_c_out])
```

I need to implement it in tensorflow 1.15, but in a way that will be compatible with tflite 1.15.
** It means that I cannot use `keras.layers.LSTM` because it is not compatible with tflite 1.15. **

Following this examples, I saw the tutorial: https://github.com/tensorflow/tensorflow/tree/r1.15/tensorflow/lite/experimental/examples/lstm
https://github.com/tensorflow/tensorflow/blob/r1.15/tensorflow/lite/experimental/examples/lstm/TensorFlowLite_LSTM_Keras_Tutorial.ipynb 

Which explains how to implement LSTM in a way that is compatible with tflite 1.15. 
I understand I need to use the following layers: `tf.lite.experimental.nn.TFLiteLSTMCell, tf.lite.experimental.nn.dynamic_rnn`

The hard part is this line:
```python
x1, state_1_h_out, state_1_c_out = layers.LSTM(200, return_sequences=True, input_shape=(sequence_length, 52),
                                               return_state=True)(inputs, initial_state=[state_1_h, state_1_c])

```

How can I implement it? 

The [dynamic_rnn documentation](https://www.tensorflow.org/api_docs/python/tf/compat/v1/lite/experimental/nn/dynamic_rnn) explains how to provide initial state to the dynamic rnn. 

I try to use it in the `buildLstmLayer` function provided (that should implement LSTM): 

```python
def buildLstmLayer(inputs, num_layers, num_units):
  """Build the lstm layer.

  Args:
    inputs: The input data.
    num_layers: How many LSTM layers do we want.
    num_units: The unmber of hidden units in the LSTM cell.
  """
  lstm_cells = []
  for i in range(num_layers):
    lstm_cells.append(
        tf.lite.experimental.nn.TFLiteLSTMCell(
            num_units, forget_bias=0, name='rnn{}'.format(i)))
  lstm_layers = tf.keras.layers.StackedRNNCells(lstm_cells)
  # Assume the input is sized as [batch, time, input_size], then we're going
  # to transpose to be time-majored.
  transposed_inputs = tf.transpose(
      inputs, perm=[1, 0, 2])
  outputs, _ = tf.lite.experimental.nn.dynamic_rnn(
      lstm_layers,
      transposed_inputs,
      dtype='float32',
      time_major=True)
  unstacked_outputs = tf.unstack(outputs, axis=0)
  return unstacked_outputs[-1]
```

I try to take this code:
```python
  outputs, _ = tf.lite.experimental.nn.dynamic_rnn(
      lstm_layers,
      transposed_inputs,
      dtype='float32',
      time_major=True)
```
And add initial state.

This try
  ```python
outputs, _ = tf.lite.experimental.nn.dynamic_rnn(
      lstm_layers,
      transposed_inputs,
      dtype='float32',
      time_major=True,
      initial_state=[tf.compat.v1.placeholder(tf.float32, shape=(200)), tf.compat.v1.placeholder(tf.float32, shape=(200))])
```
returns error `ValueError: Shape must be rank 2 but is rank 1 for 'stacked_rnn_cells/concat' (op: 'ConcatV2') with input shapes: [?,28], [200], [].
`

This try:
```python
  outputs, _ = tf.lite.experimental.nn.dynamic_rnn(
      lstm_layers,
      transposed_inputs,
      dtype='float32',
      time_major=True,
      initial_state=[tf.compat.v1.placeholder(tf.float32, shape=(1, 200)), tf.compat.v1.placeholder(tf.float32, shape=(1, 200))])
```
returns error `ValueError: Dimensions must be equal, but are 228 and 92 for 'stacked_rnn_cells/MatMul' (op: 'MatMul') with input shapes: [1,228], [64,92].
`

How can I solve it?

Thank you

**Comment by amahendrakar** [comment]
@yonatanbitton,
This question is better asked on [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow) since it is not a TensorFlow bug or feature request. There is a larger community that reads questions there. 

Also, TensorFlow 1.x is not actively supported. I'd suggest upgrading TensorFlow to the latest stable version v2.4.1. Thanks!
